db/migrate/20150613202109_create_o_auth_applications.rb

```diff
@@ -0,0 +1,17 @@
+class CreateOAuthApplications < ActiveRecord::Migration
+  def self.up
+    create_table :o_auth_applications do |t|
+      t.belongs_to :user, index: true
+      t.string :client_id
+      t.string :client_secret
+      t.string :name
+      t.string :redirect_uris
+
+      t.timestamps null: false
+    end
+  end
+
+  def self.down
+    drop_table :o_auth_applications
+  end
```

Comment:
AR is smart here, use `def change` instead of `def self.up` and it'll autogenerate you this simple down method.